### PR TITLE
Feature/composite

### PR DESCRIPTION
**Pull request by ericlp**

### CODE REVIEW

File: src/main/java/tool/designpatterns/verifiers/VerifierUtils.java [hunk 1]
```diff
@@ -141,4 +141,41 @@ public static Feedback hasFieldOfType(
...
+     *
+     * @param method1 a method
+     * @param method2 a method
+     *
+     * @return a boolean
+     */
+    public static boolean isSameMethod(MethodDeclaration method1, MethodDeclaration method2) {
+        boolean hasSameName = method1.getName().equals(method2.getName());
+        boolean hasSameParameters = method1.getParameters().equals(method2.getParameters());
+        return hasSameName && hasSameParameters;
```
Comment (ViddeM) [inline src/main/java/tool/designpatterns/verifiers/VerifierUtils.java:180]: Shouldn't this be done with the symbolsolver?

Comment (ViddeM) [reply src/main/java/tool/designpatterns/verifiers/VerifierUtils.java:178]:
```suggestion
        String qualifiedName1 = method1.resolve().getQualifiedName();
        String qualifiedName2 = method2.resolve().getQualifiedName();
        return qualifiedName1.equals(qualifiedName2);
```

File: src/main/java/tool/designpatterns/verifiers/multiclassverifiers/compositeverifier/CompositePatternInstance.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                        identifiedElements.add(container);
+                    }
+                });
+            });
+            leaves.forEach(leaf -> {
+                List<ClassOrInterfaceType> types = new ArrayList<>();
+                types.addAll(leaf.getExtendedTypes());
+                types.addAll(leaf.getImplementedTypes());
+                types.forEach(classOrInterface -> {
+                    if (classOrInterface.getNameAsString().equals(componentName)) {
```
Comment (ViddeM) [inline src/main/java/tool/designpatterns/verifiers/multiclassverifiers/compositeverifier/CompositePatternInstance.java]: Symbolsolver?

File: src/main/java/tool/designpatterns/verifiers/multiclassverifiers/compositeverifier/CompositePatternInstance.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        // Found no better way to populate PatternInstances than looping
+        // through everything multiple times
+        components.forEach((component) -> {
+            String componentName = component.getNameAsString();
+            containers.forEach(container -> {
+                List<ClassOrInterfaceType> types = new ArrayList<>();
+                types.addAll(container.getExtendedTypes());
+                types.addAll(container.getImplementedTypes());
+                types.forEach(classOrInterface -> {
+                    if (classOrInterface.getNameAsString().equals(componentName)) {
```
Comment (ViddeM) [inline src/main/java/tool/designpatterns/verifiers/multiclassverifiers/compositeverifier/CompositePatternInstance.java]: Symbolsolver?

**Review by ViddeM — DISMISSED** [review]
Look at my suggestions please.

File: src/main/java/tool/designpatterns/verifiers/VerifierUtils.java [hunk 4]
```diff
@@ -141,4 +141,41 @@ public static Feedback hasFieldOfType(
...
+     * As getQualifiedName returns the package path to the method, it cant be used to determine if
+     * two merhods are the same. Instead, we check that they have the same name and idendentical
+     * argments.
+     *
+     * @param method1 a method
+     * @param method2 a method
+     *
+     * @return a boolean
+     */
+    public static boolean isSameMethod(MethodDeclaration method1, MethodDeclaration method2) {
```
Comment (ViddeM) [inline src/main/java/tool/designpatterns/verifiers/VerifierUtils.java]: Please rename this method to something, currently I read this as that they should be the exact same method.

File: src/main/java/tool/designpatterns/verifiers/VerifierUtils.java [hunk 5]
```diff
@@ -141,4 +141,41 @@ public static Feedback hasFieldOfType(
...
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * As getQualifiedName returns the package path to the method, it cant be used to determine if
+     * two merhods are the same. Instead, we check that they have the same name and idendentical
+     * argments.
```
Comment (ViddeM) [inline src/main/java/tool/designpatterns/verifiers/VerifierUtils.java]: This should not be a javadoc, method javadocs should describe the method. This should instead be a normal java comment.

File: src/main/java/tool/designpatterns/verifiers/VerifierUtils.java (hunk 1 above)
Comment (ericlp) [reply src/main/java/tool/designpatterns/verifiers/VerifierUtils.java:180]: no is wrong. But is change name

File: src/main/java/tool/designpatterns/verifiers/multiclassverifiers/compositeverifier/CompositePatternInstance.java (hunk 3 above)
Comment (ericlp) [reply src/main/java/tool/designpatterns/verifiers/multiclassverifiers/compositeverifier/CompositePatternInstance.java]: yes

File: src/main/java/tool/designpatterns/verifiers/multiclassverifiers/compositeverifier/CompositePatternInstance.java (hunk 2 above)
Comment (ericlp) [reply src/main/java/tool/designpatterns/verifiers/multiclassverifiers/compositeverifier/CompositePatternInstance.java]: yes

**Review by ViddeM — DISMISSED** [review]

**Review by ViddeM — DISMISSED** [review]

**Review by ViddeM — APPROVED** [review]

**Review by MVacante — APPROVED** [review]

**Pull request merged**